src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+    for symbol in target.symbols() {
+        match symbol {
+            SymbolType::Main(symbol)          => scene.views.main.add(&symbol),
+            SymbolType::Visualisation(symbol) => scene.views.visualisation.add(&symbol),
+        }
+    }
+}
+
+/// Moves the given components shapes to the fullscreen scene layers.
+// FIXME This is an ugly hack for layer management.
+// FIXME Needs to be removed as soon as we have something better.
+pub fn set_layers_fullscreen<T: NativeComponent>(target:&T, scene:&Scene) {
+    target.unset_layers_all(&scene);
+    for symbol in target.symbols() {
+        match symbol {
+            SymbolType::Main(symbol)          => scene.views.overlay.add(&symbol) ,
+            SymbolType::Visualisation(symbol) => scene.views.overlay_visualisation.add(&symbol) ,
+        }
+    }
+}
+
+
+
+// ===========================
+// === Fullscreen Operator ===
+// ===========================
+
+pub trait Fullscreenable = display::Object+Resizable+NativeComponent+Networked+CloneRef+'static;
+
+/// A `FullscreenOperator` can be used to apple fullscreen mode to a UI element as well as undo the
+/// undo the fullscreen operation and restore the previous state. The  `FullscreenOperator` can be
+/// applied to any target that implements `display::Object`, `Resizable` and `NativeUiElement`.
+// TODO consider incorporating these traits into display::Object or another common "SceneElement"
+// type. But it is important that complex UI components can provide information about their
+// sub-components (for example, multiple sub-shapes or HTML components).
+#[derive(Debug)]
+pub struct FullscreenOperator<T> {
+    target            : T,
+    scene             : Scene,
+    size_original     : Vector3<f32>,
+    position_original : Vector3<f32>,
+    parent_original   : Option<display::object::Instance>,
+}
+
+impl<T:Fullscreenable> FullscreenOperator<T> {
+    /// Make the provided target fullscreen within the given scene and return the
+    /// `FullscreenOperator`.
+    pub fn apply(target:T, scene:Scene) -> Self {
+        let size_original     = target.size();
+        let position_original = target.position();
+        let parent_original   = target.display_object().rc.parent();
+        FullscreenOperator {target,scene,size_original,position_original,parent_original}.init()
+    }
+
+    fn init(self) -> Self {
+        let original_pos = self.target.display_object().global_position();
+
+        // Change parent
+        self.target.display_object().set_parent(self.scene.display_object());
+        self.target.unset_layers_all(&self.scene);
+        set_layers_fullscreen(&self.target, &self.scene);
+
+        let margin = 0.0;
+        let scene_shape = self.scene.shape();
+        let size_new    = Vector3::new(scene_shape.width(), scene_shape.height(),0.0) * (1.0 - margin);
+
+        // TODO Currently we assume objects are center aligned, but this needs to be properly
+        // accounted for here.
```

Comment:
What does "properly accounted" mean here? And what are the "objects"? When can objects not be center aligned? I mean - I really do not understanding this TODO.